frontend/integration-tests/tests/kubevirt/node.maintenance.scenario.ts

```diff
@@ -0,0 +1,83 @@
+import { waitForCount, removeLeakedResources, VM_BOOTUP_TIMEOUT, PAGE_LOAD_TIMEOUT, POD_START_TIMEOUT } from './utils';
+import { examplePod } from './mocks';
+import { appHost, testName } from '../../protractor.conf';
+import { isLoaded, filterForName, resourceRows } from '../../views/crud.view';
+import { overviewTab } from '../../views/kubevirt/virtualMachine.view';
+import { detailViewAction } from '../../views/kubevirt/vm.actions.view';
+import Pod from './models/pod';
+import * as podView from '../../views/kubevirt/pod.view';
+
+
+describe('Test Node Maintenance Mode', () => {
+  const leakedResources = new Set<string>();
+  const podName = `maintenance-${testName}`;
+  const pod = new Pod(podName, testName);
+  let computeNodeHostname = '';
+  let computeNodeURL = '';
+
+  beforeAll(async() => {
+    // Create an example hello-world pod
+    execSync(`echo '${JSON.stringify(examplePod(podName, {}))}' | kubectl create -f -`);
+    leakedResources.add(JSON.stringify({name: podName, namespace: `${testName}`, kind: 'pods'}));
+
+    await pod.navigateToTab(overviewTab);
+    await pod.waitForStatusIcon(podView.statusIcons.running, POD_START_TIMEOUT);
+
+    // store hostname of the compute node
+    computeNodeHostname = (await podView.nodeLink.getText()).split('.')[0];
+
+    await podView.nodeLink.click();
+    await isLoaded();
+
+    // store url of the compute node
+    computeNodeURL = await browser.getCurrentUrl();
+  });
+
+  beforeEach(async() => {
+    await browser.get(computeNodeURL);
+    await isLoaded;
+    await detailViewAction('Start Maintenance', true);
+  });
+
+  afterEach(async() => {
+    await browser.get(computeNodeURL);
+    await isLoaded;
+    await detailViewAction('Stop Maintenance', false);
+
+    removeLeakedResources(leakedResources);
+  });
+
+  it('Terminates running pod on maintenance node', async() => {
+    await pod.navigateToTab(overviewTab);
+    await pod.waitForStatusIcon(podView.statusIcons.terminating, PAGE_LOAD_TIMEOUT);
+
+    await browser.get(`${appHost}/k8s/ns/${testName}/pods`);
+    await isLoaded();
+
+    await filterForName(podName);
+    await browser.wait(until.and(waitForCount(resourceRows, 0)), VM_BOOTUP_TIMEOUT);
+    leakedResources.delete(JSON.stringify({name: podName, namespace: `${testName}`, kind: 'pods'}));
+  }, VM_BOOTUP_TIMEOUT);
+
+  it('Node in Maintenance mode is not schedulable', async() => {
+    execSync(`echo '${JSON.stringify(examplePod(podName, {'kubernetes.io/hostname': computeNodeHostname}))}' | kubectl create -f -`);
+    leakedResources.add(JSON.stringify({name: `${pod.name}`, namespace: `${pod.namespace}`, kind: `${pod.kind}`}));
+    await pod.navigateToTab(overviewTab);
+
+    let errorMessage = '';
+    try {
+      await pod.waitForStatusIcon(podView.statusIcons.running, POD_START_TIMEOUT);
+    } catch (error) {
+      errorMessage = error.message;
+    }
+    expect(errorMessage).toBe('Time out error');
+
+    await pod.action('Delete');
```

Comment:
I would rather check CLI here or in pod.action than wait for redirect. It is a bit safer.

Reply:
Changed this so that it's being removed by cli as it's not really necessary to remove the pod in UI + improved the delete action so that it waits until pod actually disappears in UI